OpenSim/Simulation/Model/Model.cpp

```diff
@@ -1890,8 +1892,12 @@ void Model::formStateStorage(const Storage& originalStorage, Storage& statesStor
         }
         statesStorage.append(*stateVec);
     }
-    rStateNames.insert(0, "time");
-    statesStorage.setColumnLabels(rStateNames);
+    rStateNames.insert(rStateNames.begin(), "time");
+    OpenSim::Array<std::string> rStateNamesArray;
+    for (auto const& name : rStateNames) {
+        rStateNamesArray.append(name);
+    }
+    statesStorage.setColumnLabels(rStateNamesArray);
```

Comment:
Could add a method to `OpenSim::Array` that accepts `std::vector<std::string>` for column-labels. It avoids repeated use of this snippet.